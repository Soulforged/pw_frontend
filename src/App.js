//@flow
import React from "react";
import { Provider, connect } from "react-redux";
import "src/App.css";
import { ErrorBoundary } from "src/error";
import { Login } from "src/session";
import { Home } from "src/home";
import { Loading } from "src/components";
import { ConnectedRouter } from "react-router-redux";
import { Route, Switch } from "react-router-dom";
import SecureRoute from "src/SecureRoute";
import createStore from "src/reduxconf";
import createHistory from "history/createBrowserHistory";

import type { Store } from "src/types";

type Props = {
  store?: Store<*, *>,
  history?: Object
};

const ThemeLinkC = ({ dark }: { dark: boolean }) => (
  dark ? <link rel="stylesheet" type="text/css" href="/AppDark.css" /> : false
);

const ThemeLink = connect(({ ui: { dark } }) => ({ dark }))(ThemeLinkC);

const App = ({ store, history }: Props) => (
  <Provider store={store}>
    <div>
      <ThemeLink />
      <ErrorBoundary>
        <ConnectedRouter history={history}>
          <div>
            <Switch>
              <Route path="/login" component={Login} />
              <SecureRoute path="/" component={Home} />
            </Switch>
            <Loading />
          </div>
        </ConnectedRouter>
      </ErrorBoundary>
    </div>
  </Provider>
);

const defaultHistory = createHistory();
App.defaultProps = {
  history: defaultHistory,
  store: createStore({ history: defaultHistory, logActions: process.env.DEBUG_APP })
};

export default App;
