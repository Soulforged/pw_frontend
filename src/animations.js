//@flow

export const slideTransition = (duration, key, initialValue, endValue, state) => {
  const transitions = {
    entering: { [key]: initialValue, display: "block", opacity: 0 },
    entered: { [key]: endValue, display: "block", opacity: 1 },
  };
  return {
    transition: `opacity ${key} ${duration}ms ease-in-out`,
    [key]: initialValue,
    display: "none",
    opacity: 0,
    ...transitions[state]
  };
};

export const shirnkTransition = (duration, key, initialValue, endValue, state) => {
  const transitions = {
    entering: { [key]: initialValue },
    entered: { [key]: endValue },
  };
  return {
    transition: `${key} ${duration}ms ease-in-out`,
    [key]: initialValue,
    overflow: "hidden",
    ...transitions[state]
  };
};
