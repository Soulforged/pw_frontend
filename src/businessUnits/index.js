//@flow

import BusinessUnits from "./containers/BusinessUnitContainer";
import BusinessUnitDetail from "./containers/BusinessUnitDetailContainer";
import BusinessUnitForm from "./containers/BusinessUnitFormContainer";

export { BusinessUnits, BusinessUnitDetail, BusinessUnitForm };
