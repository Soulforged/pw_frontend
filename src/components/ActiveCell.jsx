//@flow
import React from "react";

type status = 1 | 0;

export default ({ value }: { value: status }) => (
  <span className={value ? "td-success" : "td-error"}>
    <i className="fa fa-circle" />{value ? "Active" : "Inactive"}
  </span>
);
