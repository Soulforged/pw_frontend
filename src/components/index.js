//@flow

import Loading from "./Loading";

export { Loading }; //eslint-disable-line
