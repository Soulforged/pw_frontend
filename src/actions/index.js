//@flow
import { createActions } from "redux-actions";
import { CALL_API } from "src/constants";
import type { UserInfo, AppError } from "src/types";

const ENTITIES = "@@entities";
const SAVE = "SAVE";
const REQ = "REQUEST";
const SUCCESS = "SUCCESS";
const FAIL = "FAILURE";
const LOGIN_SUCCESS = "LOGIN_SUCCESS";
const LOGIN_REQUEST = "LOGIN_REQUEST";
const LOGIN_FAILURE = "LOGIN_FAILURE";

const saveTypes = (actionName) => {
  const normActionName = actionName.toUpperCase();
  return [
    `${ENTITIES}/${SAVE}_${normActionName}_${REQ}`,
    `${ENTITIES}/${SAVE}_${normActionName}_${SUCCESS}`,
    `${ENTITIES}/${SAVE}_${normActionName}_${FAIL}`
  ];
};

const fetchTypes = actionName => [
  `${actionName}_${REQ}`,
  `${actionName}_${SUCCESS}`,
  `${actionName}_${FAIL}`
];

const fetchEntitiesTypes = actionName => fetchTypes(`${ENTITIES}/${actionName.toUpperCase()}`);

const login = ({ username, password }: UserInfo) => ({
  [CALL_API]: {
    types: [LOGIN_REQUEST, LOGIN_SUCCESS, LOGIN_FAILURE],
    endpoint: "/authenticate/generate-token",
    body: { user: username, password }
  }
});

const syncActions = createActions({
  SET_ERROR: (error: AppError) => error,
  LOGOUT: () => ({}),
  TOGGLE_THEME: () => ({}),
  TOGGLE_MENU: () => ({}),
  TOGGLE_USER_MENU: () => ({}),
});

export default {
  ENTITIES,
  SAVE,
  REQ,
  SUCCESS,
  FAIL,
  saveTypes,
  fetchTypes,
  fetchEntitiesTypes,
  LOGIN_SUCCESS,
  LOGIN_FAILURE,
  LOGIN_REQUEST,
  login,
  ...syncActions
};
