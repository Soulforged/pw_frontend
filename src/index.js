import React from "react";
import ReactDOM from "react-dom";
import "bootstrap/dist/css/bootstrap.min.css";
import "font-awesome/css/font-awesome.css";
import "react-table/react-table.css";
import "src/index.css";
import App from "src/App";
import registerServiceWorker from "src/registerServiceWorker";

ReactDOM.render(<App />, document.getElementById("root"));
registerServiceWorker();
