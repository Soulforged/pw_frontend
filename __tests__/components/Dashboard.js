//@flow
import React from "react";
import { Loading } from "src/components";
import renderer from "react-test-renderer";

it("matches loading snapshot", () => {
  const render = renderer.create(<Loading />).toJSON();
  expect(render).toMatchSnapshot();
});

it("matches not loading snapshot", () => {
  const render = renderer.create(<Loading loading={false} />).toJSON();
  expect(render).toMatchSnapshot();
});
