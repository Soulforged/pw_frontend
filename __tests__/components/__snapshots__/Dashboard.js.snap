// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches loading snapshot 1`] = `
<div
  id="loader"
  style={Object {}}
>
  <img
    alt="loading something..."
    className="loader"
    src="loader.gif"
  />
</div>
`;

exports[`matches not loading snapshot 1`] = `null`;
